.github/workflows/reviewdog.yml

```
@@ -0,0 +1,104 @@
+name: reviewdog
+on:
+  pull_request:
+    branches: [ master ]
+  push:
+    branches: [ master ]
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    steps:
+      - if: github.event_name == 'push'
+        run: |
+          echo "REPORTER='github-check'" >> $GITHUB_ENV
+      - if: github.event_name == 'push'
+        run: |
+          echo "FILTER='nofilter'" >> $GITHUB_ENV
+      - if: github.event_name == 'push'
+        run: |
+          echo "FAIL='false'" >> $GITHUB_ENV
+      - if: github.event_name == 'pull_request'
+        run: |
+          echo "REPORTER=github-pr-review" >> $GITHUB_ENV
+      - if: github.event_name == 'pull_request'
+        run: |
+          echo "FILTER='added'" >> $GITHUB_ENV
+      - if: github.event_name == 'pull_request'
+        run: |
+          echo "FAIL='true'" >> $GITHUB_ENV
```

```suggestion
      - if: github.event_name == 'push'
        run: |
          echo "REPORTER='github-check'" >> $GITHUB_ENV
          echo "FILTER='nofilter'" >> $GITHUB_ENV
          echo "FAIL='false'" >> $GITHUB_ENV
      - if: github.event_name == 'pull_request'
        run: |
          echo "REPORTER=github-pr-review" >> $GITHUB_ENV
          echo "FILTER='added'" >> $GITHUB_ENV
          echo "FAIL='true'" >> $GITHUB_ENV
```
(probably does the same thing and fails but just a suggestion separately)